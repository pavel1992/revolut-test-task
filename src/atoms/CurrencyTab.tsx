import styled from 'styled-components';

import { colors } from '../styleConstants/colors';
import { TABLET_BREAKPOINT } from '../styleConstants/mediaConstants';

export interface CurrencyTabProps {
    active: boolean;
}

export const CurrencyTab = styled.div<CurrencyTabProps>`
  line-height: 25px;
  cursor: pointer;
  padding-bottom: 10px;
  color: ${props => props.active ? colors.black : colors.disabledGray};
  border-bottom: 2px solid ${props => props.active ? colors.pink : 'transparent'};
  margin-right: 24px;

  @media(max-width: ${TABLET_BREAKPOINT}) {
    padding: 8px 0 8px 8px;
    margin-right: 0;
    margin-bottom: 4px;
    border-bottom: 0;
    border-left: 2px solid ${props => props.active ? colors.pink : 'transparent'};
  }
`;

CurrencyTab.displayName = 'CurrencyTab';
