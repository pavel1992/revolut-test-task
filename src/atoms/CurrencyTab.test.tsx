import * as React from 'react';
import renderer from 'react-test-renderer';

import { CurrencyTab } from './CurrencyTab';


test('renders CurrencyTab correctly', () => {
  const render = renderer
    .create(<CurrencyTab active={false}>111</CurrencyTab>)
    .toJSON();
  expect(render).toMatchSnapshot();
});
