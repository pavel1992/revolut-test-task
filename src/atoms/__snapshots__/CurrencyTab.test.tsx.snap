// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CurrencyTab correctly 1`] = `
<div
  className="sc-AxjAm ioRucc"
>
  111
</div>
`;
